### DISCUSSION TITLE
OAuth: Support for Dynamic Registration and / or End Session Endpoint Uri

### DISCUSSION THREAD

**Discussion by shresthajibesh** [firefly-iii/firefly-iii#7631] [Firefly III]
I am working a client application that connects to Firefly using OAuth and as of now I am wondering if:

1. Do we have support of dynamic registration: As of now we can either cache and use Client Secret or use personal access token, I was wondering if we have endpoint for dynamic registration.
2. What is the End Session End Point URI, we have authenticate and token uri's but what about end session?

**Comment by JC5** [comment]
Sorry, I don't know what either of those are. Is it in the [docs](https://laravel.com/docs/10.x/passport)?

**Reply by shresthajibesh** [reply]
Thank you for the reply, 
Dynamic registration would be used for getting access token without client secret in the client. But the doc you linked,  mentioned PKCE so let me look into a bit more and ill update this. 

Similarly end session url is basically when user logs out we want to the clear the session info from server as well. Is there something for that?

**Reply by JC5** [reply]
No I don't think so. I believe it's built in.
